src/components/header/OverflowEdgeIndicator.js

```diff
@@ -2,21 +2,40 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+// @flow
+
 import React, { PureComponent, PropTypes } from 'react';
 import classNames from 'classnames';
 
 import './OverflowEdgeIndicator.css';
 
+type Props = {
+  className: string,
+  children: number,
+};
+
 class OverflowEdgeIndicator extends PureComponent {
-  constructor(props) {
+  props: Props;
+  state: {
+    overflowsOnTop: ?boolean,
```

Comment:
The only nitpick I have here is that the booleans here are not technically optional, as this means `boolean | void | null`, while in fact the state here is always just true or false, so it's `boolean` all by itself. I'm going to go ahead and merge it in though as it's not a huge deal.